### DISCUSSION TITLE
[HW3] prob2.2 & 2.3

### DISCUSSION THREAD

**Discussion by emjay73** [seominjoon/kaist-ai605#151] [Q&A] [answered]
I just want to clarify the task.
please let me know if there's a misunderstanding.

## prob2.2
**vocab**
characters
**train dataset example**
helao -> hello
**eval example**
"helvo my namw is blabla" -> "hello my namw is blabla"
do this word by word
**accuracy**
count the number of fully corrected sentences and divide by the total number of sentences.
ex) "hello my namw is blabla" is a false answer cuz of "namw"
ex) "hello my name is blabla" is a true answer cuz it's fully corrected.

## prob2.3
**vocab**
characters
**train dataset example**
"helvo my namw is blabla" -> "hello my name is blabla"
**eval example**
"helvo my namw is blabla" -> "hello my name is blabla"
do this inference at once using input sentence
**compare**
prob2.2 eval example & prob2.3 eval example
show the case when prob2.3 eval example output is better than the prob2.2's.

Is that correct?

**Comment by sorobedio** [comment, marked as answer]
According to what i understood in the  professor to  #142  
problem 2.2  the model input is one word at a time and outputs the corrected  form is needed ( i think that it is mentioned in the description that the model receives a word not a sentence in that case)
problem 2.3. the input is a sentence (sequence of words) with some incorrect then outputs the corrected sentence.

this is who i see the problem.

**Reply by emjay73** [reply]
Thank you for your answer. 
I'm still confused because of this sentence in the prob2.3.
```
find 3 examples where the word-level model fails and sentence-level model correctly predicts.
```
In order to compare both models, I thought both should be able to be applied to a sentence.

**Reply by ssong18** [reply]
The way I understood the problem: 

Have a single word with errors for both questions. 

For example, you had 'name' misspelled as 'namw' in your example. 
With the model from prob2.2, correct 'namw' to 'name'
With the model from prob2.3, correct 'hello my namw is blabla' to 'hello my name is blabla'

Reading the question again for prob2.2, 
> Your job is to create a model whose input is **_a word_** with spelling errors, and the output is the spelling-corrected word. 

So the goal is to see if the model form prob2.3 could correct words that were not corrected with the model from prob2.2, and present those three examples.

**Reply by seominjoon** [reply]
@ssong18 is correct.

**Reply by emjay73** [reply]
A-HA
I totally misunderstood the question and tried multiple stupid attempts.
Thank you..
